command/pki_reissue_intermediate.go

```diff
@@ -0,0 +1,297 @@
+		Completion: complete.PredictSet("internal", "existing", "kms"),
+	})
+
+	f.StringVar(&StringVar{
+		Name:    "issuer_name",
+		Target:  &c.flagNewIssuerName,
+		Default: "",
+		EnvVar:  "",
+		Usage:   `If present, the newly created issuer will be given this name`,
+	})
+
+	return set
+}
+
+func (c *PKIReIssueCACommand) Run(args []string) int {
+	// Parse Args
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	args = f.Args()
+
+	if len(args) < 3 {
+		c.UI.Error("Not enough arguments: expected parent issuer and child-mount location and some key_value argument")
+		return 1
+	}
+
+	stdin := (io.Reader)(os.Stdin)
+	userData, err := parseArgsData(stdin, args[3:])
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Failed to parse K=V data: %s", err))
+		return 1
+	}
+
+	// Check We Have a Client
+	client, err := c.Client()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Failed to obtain client: %v", err))
+		return 1
+	}
+
+	parentIssuer := sanitizePath(args[0]) // /pki/issuer/default
+	intermediateMount := sanitizePath(args[2])
+
+	templateCertificateResp, err := client.Logical().Read(sanitizePath(args[1]))
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error fetching template certificate %v : %v", sanitizePath(args[1]), err))
+		return 1
+	}
+	templateCertificateRaw, ok := templateCertificateResp.Data["certificate"]
+	if !ok {
+		c.UI.Error(fmt.Sprintf("No Certificate Field Found at %v instead found : %v", sanitizePath(args[1]), templateCertificateResp))
+		return 1
+	}
+	certificatePemString := templateCertificateRaw.(string)
+	certificatePem := []byte(certificatePemString)
+	certificateBlock, _ := pem.Decode(certificatePem)
+	certificate, err := x509.ParseCertificate(certificateBlock.Bytes)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error parsing template certificate at %v : %v", sanitizePath(args[1]), err))
+		return 1
+	}
+
+	useExistingKey := c.flagKeyStorageSource == "existing"
+	keyRef := ""
+	if useExistingKey { // TODO: Better Error information
+		keyRef = templateCertificateResp.Data["key_id"].(string)
+	}
+
+	templateData, err := parseTemplateCertificate(*certificate, useExistingKey, keyRef)
+	data := updateTemplateWithData(templateData, userData)
+
+	return pkiIssue(c.BaseCommand, parentIssuer, intermediateMount, c.flagNewIssuerName, c.flagKeyStorageSource, data)
+}
+
+func updateTemplateWithData(template map[string]interface{}, changes map[string]interface{}) map[string]interface{} {
+	data := map[string]interface{}{}
+
+	for key, value := range template {
+		data[key] = value
+	}
+
+	// ttl and not_after set the same thing.  Delete template ttl if using not_after:
+	if _, ok := changes["not_after"]; ok {
+		delete(data, "ttl")
+	}
+
+	// If we are updating the key_type, do not set key_bits
+	if _, ok := changes["key_type"]; ok && changes["key_type"] != template["key_type"] {
+		delete(data, "key_bits")
```

Comment:
perhaps `signature_bits` and `use_pss` should be reset here as well if they changed the key_type on us?

Reply:
signature_bits and use_pss refer to the signer (parent?) key-type rather than the child (newly created issuer) key bits.  So the child key-type changing I don't think affects this.  I'll add a check to test whether this breaks anything if we use a new parent who isn't using rsa though.

Reply:
(Test is updated) https://github.com/hashicorp/vault/pull/18499/commits/9a7b5071dfbd38d787eaaa2cc20a419b8f707ac2

Reply:
Yes you are absolutely correct, *sigh* Friday afternoon reviews...